hudi-client/hudi-client-common/src/main/java/org/apache/hudi/config/HoodieCleanConfig.java

```diff
@@ -77,9 +80,9 @@ public class HoodieCleanConfig extends HoodieConfig {
       .withDocumentation(HoodieCleaningPolicy.class)
       .markAdvanced()
       .withInferFunction(cfg -> {
-        boolean isCommitsRetainedConfigured = cfg.contains(CLEANER_COMMITS_RETAINED_KEY);
-        boolean isHoursRetainedConfigured = cfg.contains(CLEANER_HOURS_RETAINED_KEY);
-        boolean isFileVersionsRetainedConfigured = cfg.contains(CLEANER_FILE_VERSIONS_RETAINED_KEY);
+        boolean isCommitsRetainedConfigured = cfg.contains(CLEANER_COMMITS_RETAINED_KEY) || cfg.contains(CLEANER_COMMITS_RETAINED_ALTERNATIVE_KEY);
```

Comment:
if you have added alternatives below, config.contains should take care of it. 
```   public static <T> boolean contains(ConfigProperty<T> configProperty, HoodieConfig config) {
    if (config.getProps().containsKey(configProperty.key())) {
      return true;
    }
    return configProperty.getAlternatives().stream().anyMatch(k -> config.getProps().containsKey(k));
  }
```

So, we don't need to fix this on top of it.

Reply:
Actually this is necessary because the order the fields are declared impacts how the defaults are computed. The other config properties need to be defined below this or else the defaults will be added to the properties and interfere with this default calculation.

Reply:
We should revisit the impact of ordering of config definition for inferring the config values in a follow-up (HUDI-9153), as this pattern is used in other places and can be disturbed easily by changing the order.